submissions/zalevskyi/a-tiny-js-world-oop/index.js

```diff
@@ -0,0 +1,148 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+
+   Code repository: https://github.com/zalevskyi/kottans-frontend/tree/main/practice/a-tiny-js-world
+   Web app: https://zalevskyi.github.io/kottans-frontend/practice/a-tiny-js-world/index.html
+   */
+
+// ======== OBJECTS DEFINITIONS ========
+// Define your objects here
+
+class Species {
+   constructor(genus) {
+      this.genus=genus
+   }
+   descriptionArray() {
+      return [`species: ${this.genus}`]
+   }
+}
+class MammalNonMarine extends Species {
+   constructor(genus) {
+      super(genus)
+      this.legs = 4
+   }
+   descriptionArray() {
+      return [...super.descriptionArray(), `legs: ${this.legs}`]
+   }
+}
+class CarnivoraNonMarine extends MammalNonMarine {
+   constructor(genus, sound) {
+      super(genus)
+      this.sound = sound
+   }
+   descriptionArray() {
+      return [...super.descriptionArray(), `sound: ${this.sound}`]
+   }
+}
+class Hominoid extends MammalNonMarine {
+   constructor(genus, sound) {
+      super(genus)
+      this.legs = 2 // overides property value of the prototype
+      this.hands = 2
+      this.sound = sound
```

Comment:
Both Hominoid and CarnivoraNonMarine have `sound`, which means it is a shared property and thus should be generalized to their common ancestor.